### DISCUSSION TITLE
Filter query

### DISCUSSION THREAD

**Discussion by harshitsinghai77** [deta/cloud-docs#252] [Help] [answered]
This is how my deta looks like

![image](https://user-images.githubusercontent.com/30886142/172084885-66c7690b-d28b-440e-8fd5-ea86e600654f.png)

I want to query all the entries that have a `created_at` in the last 7 days. But my queries return an empty list.

```python
query = {"key": user_id, "analytics.duration?lt": str(datetime.now())}
```

I even tried, this is also empty,

```python
query = {"key": user_id, "analytics.duration": 4501}
```

what am I doing wrong?

**Comment by jnsougata** [comment]
`analytics` is a list so you have to do `analytics[0].duration?lt: str(datetime.now())`

The way you are storing the data is not much efficient to make an exact query. 
Rather store the data with individual user/keys. It will make extraction easy and faster and more efficient. 
You can see table below for reference:
![image](https://user-images.githubusercontent.com/53375272/172090690-184d2b43-1683-492b-9469-e763407b953c.png)
and there are two attributes to the user anlytics `clicks` and `retention` and stored as the table below:
![image](https://user-images.githubusercontent.com/53375272/172090967-da13c5a3-e8b9-43ad-bb34-3c6a186f214d.png)
and the query for all the users having `clicks > 800` will be `{'analytics.clicks?gt': 800}`
and you will receive a response as:
`{'paging': {'size': 2}, 'items': [{'analytics': {'clicks': 900, 'retention': 0.15}, 'key': 'user_8'}, {'analytics': {'clicks': 1000, 'retention': 0.16666666666666666}, 'key': 'user_9'}]}`

**Comment by abdelhai** [comment, marked as answer]
i hate to say it but you are using base wrong. for queries to work and work efficiently you need to plan out your schemas etc.
i don't recommend putting all user events in the user object. the **correct approach** is to use a separate base for events and another one for users.

**Reply by jnsougata** [reply]
If separate base is assigned for each individual user it will be hard to implement caching locally. So I prefer the single base pattern where I can cache the whole base at start up and update in realtime. In this way I can reduce API calls as well.

**Reply by abdelhai** [reply]
i didn't mean a base for each user, rather one base for all events and one base for all users.

**Reply by jnsougata** [reply]
oh mb

**Reply by harshitsinghai77** [reply]
Got it. Thanks for the help.
